src/plans/create/steps/network-mappings/NetworkMapFieldTable.tsx

```diff
@@ -0,0 +1,157 @@
+import { type FC, useCallback } from 'react';
+import { type FieldPath, useFieldArray } from 'react-hook-form';
+
+import { Button, ButtonVariant, Icon, Stack } from '@patternfly/react-core';
+import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import type { CreatePlanFormData } from '../../constants';
+import { useCreatePlanFormContext } from '../../hooks';
+
+import {
+  defaultNetMapping,
+  netMapFieldLabels,
+  NetworkMapFieldId,
+  type NetworkMapping,
+} from './constants';
+import SourceNetworkField from './SourceNetworkField';
+import TargetNetworkField from './TargetNetworkField';
+import { getNetworkMapFieldId } from './utils';
+
+import './NetworkMapFieldTable.style.scss';
+
+type NetworkMapFieldTableProps = {
+  targetNetworks: Record<string, string>;
+  usedSourceLabels: string[];
+  otherSourceLabels: string[];
+  isLoading: boolean;
+  loadError: Error | null;
+};
+
+export const NetworkMapFieldTable: FC<NetworkMapFieldTableProps> = ({
+  isLoading,
+  loadError,
+  otherSourceLabels,
+  targetNetworks,
+  usedSourceLabels,
+}) => {
+  const { t } = useForkliftTranslation();
+  const { control, setValue } = useCreatePlanFormContext();
+
+  const validate = useCallback(
```

Comment:
seems a bit strange this function called validate, can return boolean or a string?

Reply:
Yes, it can return a boolean or a string. This is the name of the prop used in "rules" for a given controlled field using react-hook-form.